### PR TITLE
ArcSpan RTD Module: Initial Release

### PR DESCRIPTION
**Pull request by josecu**
## Type of change
- [x] Feature

## Description of change

Submitting the RTD Module for ArcSpan for initial review.

## Other information

Docs PR: https://github.com/prebid/prebid.github.io/pull/4304

### CODE REVIEW

File: modules/arcspanRtdProvider.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  name: SUBMODULE_NAME,
+  init: init,
+  getBidRequestData: alterBidRequests,
+};
+
+function init(config, userConsent) {
+  if (typeof config.params.silo === 'undefined') {
+    return false;
+  }
+  var _s = document.createElement('script');
```
Comment (patmmccann) [inline modules/arcspanRtdProvider.js:19]: please use our methods to insert scripts

Comment (patmmccann) [reply modules/arcspanRtdProvider.js:19]: also can you check for  window.arcobj1 before appending the script and recommend that it is deployed that way in your doc, you'll get better coverage that way. also, you'll want to make the location of the script configurable and fallback to a place that is locked to a version if you decide you do want prebid sourcing your location

File: modules/arcspanRtdProvider.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        keywords: _v1s.toString(),
+        content: _content,
+      },
+    };
+    mergeDeep(reqBidsConfigObj.ortb2Fragments.global, _ortb2);
+  }
+  callback();
+}
+
+submodule(MODULE_NAME, arcspanSubmodule);
```
Comment (patmmccann) [inline modules/arcspanRtdProvider.js:71]: you should make a commit to submodules.json

File: modules/arcspanRtdProvider.js [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+function init(config, userConsent) {
+  if (typeof config.params.silo === 'undefined') {
+    return false;
+  }
+  var scriptUrl;
+  if (config.params.silo === 'test') {
+    scriptUrl = 'https://localhost:8080/as.js';
+  } else {
+    scriptUrl = 'https://silo' + config.params.silo + '.p7cloud.net/as.js';
```
Comment (patmmccann) [inline modules/arcspanRtdProvider.js:24]: please handle publisher hosted js here, or cases where the publisher has already loaded the js before or in parallel with prebid.js to improve performance. Please provide instructions on this, as making your js dependent on prebid.js means less auctions will contain your enrichments

**Review by patmmccann — CHANGES_REQUESTED** [review]
additional feedback on enabling parallel loading

File: modules/arcspanRtdProvider.js (hunk 1 above)
Comment (josecu) [reply modules/arcspanRtdProvider.js]: Thanks for the quick review, @patmmccann! I implemented the insert script method & the check, but I'm not sure I understand the 2nd part of your last comment. Do you mean we would have to make the location configurable and implement the fallback if/when we decide to have prebid source our location? Or do you mean we should make the location configurable and implement the fallback now, in preparation for if/when we decide to have prebid source our location?

File: modules/arcspanRtdProvider.js (hunk 3 above)
Comment (josecu) [reply modules/arcspanRtdProvider.js]: @patmmccann We don't currently support publisher-hosted JS. Is that a requirement here? I implemented the performance changes. I also added instructions to the documentation for the publisher to reference our onboarding documentation to deploy the tag outside of Prebid.js. Is that sufficient?

Comment (patmmccann) [reply modules/arcspanRtdProvider.js]: that works, ty!

**Review by patmmccann — APPROVED** [review]

**Pull request merged**